crates/store/src/genesis/config/mod.rs

```diff
@@ -0,0 +1,291 @@
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub enum AccountConfig {
+    Wallet(WalletConfig),
+    Faucet(FaucetConfig),
+}
+
+/// `false` doesn't pass the `syn::Path` parsing, so we do one level indirection
+const fn ja() -> bool {
+    true
+}
+
+/// Represents a wallet, containing a set of assets
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct WalletConfig {
+    /// Provide a name, that will be used for the keyfile.
+    #[serde(default)]
+    name: Option<String>,
+    #[serde(default)]
+    is_updatable: bool,
+    #[serde(default)]
+    storage_mode: StorageMode,
+    assets: Vec<AssetEntry>,
+}
+
+/// Represents a faucet with asset specific properties
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct FaucetConfig {
+    /// Provide a name, that will be used for the keyfile.
+    #[serde(default)]
+    name: Option<String>,
+    // TODO eventually directly parse to `TokenSymbol`
+    symbol: String,
+    decimals: u8,
+    max_supply: u64,
+    #[serde(default)]
+    storage_mode: StorageMode,
+    #[serde(default = "self::ja")]
+    fungible: bool,
+}
+
+/// See the [full description](https://0xmiden.github.io/miden-base/account.html?highlight=Accoun#account-storage-mode)
+/// for details
+#[derive(Debug, Clone, Copy, serde::Serialize, serde::Deserialize, Default)]
+pub enum StorageMode {
+    /// Monitor for `Notes` related to the account, in addition to being `Public`.
+    #[serde(alias = "network")]
+    #[default]
+    Network,
+    /// A publicly stored account, lives on-chain.
+    #[serde(alias = "public")]
+    Public,
+    /// A private account, which must be known by interactors.
+    #[serde(alias = "private")]
+    Private,
+}
+
+impl From<StorageMode> for AccountStorageMode {
+    fn from(mode: StorageMode) -> AccountStorageMode {
+        match mode {
+            StorageMode::Network => AccountStorageMode::Network,
+            StorageMode::Private => AccountStorageMode::Private,
+            StorageMode::Public => AccountStorageMode::Public,
+        }
+    }
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+struct AssetEntry {
+    symbol: String, // TODO move to a wrapper around `TokenSymbol`
+    amount: u64,    // TODO we might want to provide `humantime`-like denominations
+}
+
+#[derive(Debug, Clone)]
+pub struct AccountFileWithName {
+    pub name: String,
+    pub account_file: AccountFile,
+}
+
+/// Secrets generated during the state generation
+#[derive(Debug, Clone)]
+pub struct AccountSecrets {
+    // name, account, private key, account seed
+    pub secrets: Vec<(Option<String>, Account, SecretKey, Word)>,
+}
+
+impl AccountSecrets {
+    /// Convert the internal tuple into an `AccountFile`
+    ///
+    /// If no name is present, a new one is generated based on the current time
+    /// and the index in
+    pub fn as_account_files(&self) -> impl Iterator<Item = AccountFileWithName> {
```

Comment:
As mentioned elsewhere I'm unsure about this.

Should we not keep it simple and name them `account_{i}` with no exceptions. The configurer can rename, move or otherwise do whatever they want with them.

Using the index is correct here; but doesn't line up with the order in which they're defined in the toml so this would be extremely confusing to the configurer.

I don't think the time adds anything - if they clobber existing keys we should either let them, or panic and avoid that. Probably the latter.

Reply:
I would 
- combine the config/toml wallets into a single enum property so they can be ordered
- name the files `account_{i}`
- avoid time and just refuse to clobber existing files

Alternatively, require a file name as part of the config. But I wouldn't allow flexibility here.

Reply:
Another alternative is to name them `faucet_{symbol}` and then `wallet_{i}`? That could also be acceptable I think, and lets you order them separately still

Reply:
Implemented